lib/foreman_discovery/engine.rb

```diff
@@ -45,16 +45,23 @@ class Engine < ::Rails::Engine
             :discovered_hosts          => [:index, :show, :auto_complete_search],
             :"api/v2/discovered_hosts" => [:index, :show]
           }, :resource_type => 'Host'
-          permission :provision_discovered_hosts, {
+          permission :submit_discovered_hosts, {
+            :"api/v2/discovered_hosts" => [:facts, :create]
+          }, :resource_type => 'Host'
+          permission :auto_provision_discovered_hosts, {
             :discovered_hosts          => [:auto_provision, :auto_provision_all],
-            :"api/v2/discovered_hosts" => [:create, :auto_provision, :auto_provision_all]
+            :"api/v2/discovered_hosts" => [:auto_provision, :auto_provision_all]
+          }, :resource_type => 'Host'
```

Comment:
hm, just double check that it generates queries as expected, `Host` is proxy for `Host::Managed` but I suppose you need `Host::Discovered`. Chances are it works correctly because of some if since the same resource_type is use elsewhere but looks suspicious.